rake, spec and CentOS 7 fixes

jantman opened the pull request:
(1) Fix a number of Rake and spec issues so the tests actually __work__.

(2) Per https://support.newrelic.com/requests/130089, have Puppet manage the group and user on RedHat 7. This is because of a broken useradd command in the RPM that fails on distros using shadow-utils >= 4.1.5. I strongly disagree with fixing a broken RPM in a Puppet module, but this is what Support told me to do.

On a side note:
(1) This module is way out of date in terms of Puppet module standards. It uses a number of deprecated patterns and the spec tests are woefully incomplete. When Puppet4 comes out (which is going to be soon), this module will no longer be functional.
(2) There are a number of places for improvement, such as the exec used to install the Yum repository, when there's a native type for this.

jantman commented:
This fixes the CentOS7 issue for me.

jessedearing commented on manifests/init.pp:
```diff
@@ -121,9 +121,34 @@
+      # https://support.newrelic.com/requests/130089
+      # workaround for RPM that's broken on Cent7+ (or any other distro
+      # that uses shadow-utils >= 4.1.5 )
+      if ( $::operatingsystemmajrelease == '7' or $::operatingsystemmajrelease == 7 ) {
```
I believe the issue is specific to SELinux and not just CentOS 7. We have CentOS 7 with shadow-utils 4.1.5 machines and this works just fine.
This should test for the `$::selinux` fact instead of `$::operatingsystemmajrelease`

jessedearing commented:
Thanks for the PR and the help getting this module into shape.

We'll look at getting it ready for Puppet 4 at some point.

jantman commented:
@jessedearing ok, I've switched that conditional from RedHat 7+ to SELinux and updated the specs.

I haven't switched over to Puppet4 yet, but I might have some bandwidth to assist with this module when we started rolling Puppet4 out in our test environments.

jantman commented:
@jessedearing in our support ticket, I was told a fix for the RPM issue should be coming soon and this won't be merged. Would you like me to cut a separate PR that removes the SELinux/CentOS7 stuff, and just has the rspec and puppet-lint fixes?

jessedearing commented:
@jantman That would be great!

jantman commented:
Closing in favor of #7 which has the rakefile/testing/spec fixes but none of the CentOS7/SELinux stuff.
